xbmc/epg/EpgContainer.cpp

```diff
@@ -829,13 +829,15 @@ void CEpgContainer::UpdateEpgEvents()
   if (!m_lastEpgEventPurge.IsValid() || m_lastEpgEventPurge < (now - CDateTimeSpan(1,0,0,0)))
   {
     CDateTime purgeTime = now - CDateTimeSpan(0, g_advancedSettings.m_iEpgLingerTime / 60, g_advancedSettings.m_iEpgLingerTime % 60, 0);
-    for (auto event = m_epgEvents.begin(); event != m_epgEvents.end(); ++event)
+    for (auto event = m_epgEvents.begin(); event != m_epgEvents.end();)
     {
       if (event->second->EndAsUTC() < purgeTime)
       {
         m_epgEvents.erase(event);
         ++count;
       }
+      else
+        ++event;
```

Comment:
This will also go wrong. If I see it right you programmed an endless loop - as the event iterator is not touched. What happens after you erased the first element? You don't exit the loop and event will keep its old value and therefore the loop above will be endless, right?